packages/design/src/DataTable/useTable.ts

```diff
@@ -172,14 +173,39 @@ function searchAndFilterCb<T>(
   searchValue: string,
   propName: keyof T & string
 ) {
-  if (propName === 'tags') {
-    return targetValue.some(item => {
-      return item.toLocaleLowerCase().includes(searchValue.toLocaleLowerCase());
-    });
-  }
   if (propName.toLocaleLowerCase().includes('date')) {
     return displayDate(targetValue).includes(searchValue);
   }
+  if (propName.toLocaleLowerCase().includes('time')) {
+    return displayDateTime(targetValue).includes(searchValue);
+  }
+  if (typeof targetValue === 'boolean') {
+    return (
+      propName.toLocaleLowerCase().includes(searchValue.toLocaleLowerCase()) &&
+      targetValue
+    );
+  }
```

Comment:
i'm not sure how i feel about this and others b/c we are depending on the propName to include a certain keyword, why did we go away from sending in custom functions?

Reply:
The custom search functions led to too much repeated code